sympy/functions/elementary/piecewise.py

```diff
@@ -305,12 +306,42 @@ def doit(self, **hints):
             newargs.append((e, c))
         return self.func(*newargs)
 
+    def is_meromorphic(self, x, a):
+        """
+        This method tests whether a Piecewise expression is meromorphic
+        as a function of the given symbol ``x`` at the point ``a``.
+
+        This method is intended as a quick test that will return
+        None if no decision can be made without simplification or
+        more detailed analysis.
+
+        Examples
+        ========
+        >>> from sympy import csc, log, Piecewise
+        >>> from sympy.abc import x
+        >>> f = Piecewise((csc(1/x), x < 1), (2*x**2 + 1, x < 5), (log(x), True))
+        >>> f.is_meromorphic(x, 0)
+        False
+        >>> f.is_meromorphic(x, 1)
+        True
+        >>> f.is_meromorphic(x, 10)
+        True
+
+        """
+        if not x.is_symbol:
+            raise TypeError("{} should be of symbol type".format(x))
+        a = sympify(a)
+
+        for arg in self.args:
+            if arg[1].subs(x, a):
+                return arg[0]._eval_is_meromorphic(x, a)
+
     def _eval_simplify(self, **kwargs):
         return piecewise_simplify(self, **kwargs)
 
     def _eval_as_leading_term(self, x, logx=None, cdir=0):
         for e, c in self.args:
-            if c == True or c.subs(x, 0) == True:
+            if (cdir == -1 and c.subs(x, -1e-3)) or ((cdir == 1 or cdir == 0) and c.subs(x, 1e-3)):
```

Comment:
This should not depend on exact values like `cdir == 1`. Although such values are currently set in limit code, any positive value could be used instead. All numbers of the form `t*c` for positive `t` define the same complex direction. Also values like `-1e-3` will not work in general.

Reply:
We should probably set `cdir = S.One` in the limit code. Then the sign could be checked by `cdir.is_positive`.

Reply:
> 

Yeah agreed , I was just trying to get things working first of all and convey my approach and then implement all relevant suggestions . The basic approach was that if we consider an example as follows `expr = Piecewise((x - 3, x < 0), ((x + 1)**2, x < 10), (5, True))`

![image](https://user-images.githubusercontent.com/87052487/143773476-5fb03d53-594b-442b-94d3-16630c570d7a.png)

We want the `as_leading_term(x)` from the negative side to be `-3` given by the this piece `(x - 3, x < 0)` and `+1` from the right  given by the this piece `((x + 1)**2, x < 10)`  right ?? Now how should I approach this ? I thought of having a margin which gets us a value just lesser than `0` and a value just greater than `0`  to approach this and I ended up using `1e-3` just as a dummy 

Replying to @smichr , I thought of getting the expression from the negative side when `cdir = -1` and the following consecutive expression from the positive side when `cdir = 1` and for the same I thought of proposing a `get_cond ` method . Surely I would try out your suggestion , I was actually trying to using the` ._intervals method` sometime back and I'll try to use your soln for the same

Reply:
> 

Yupp this should surely be done